xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1386,6 +1386,19 @@ is_downgrade(dict_t *options, gf_boolean_t *downgrade)
     return ret;
 }
 
+void
+glusterd_destroy_remote_hostname_list(glusterd_conf_t *priv)
```

Comment:
This is almost identical to `glusterd_destroy_hostname_list()`. I would reuse the existing function passing the list to destroy.